spring-cloud-dataflow-docs/src/main/asciidoc/configuration.adoc

```diff
@@ -57,11 +57,103 @@ As these are Spring Boot `@ConfigurationProperties`, that you need to specify by
 
 [source,bash,subs=attributes]
 ----
-$ SPRING_APPLICATION_JSON='{ "maven": { "local-repository": null,
-"remote-repositories": { "repo1": { "url": "https://repo1", "auth": { "username": "repo1user", "password": "repo1pass" } }, "repo2": { "url": "https://repo2" } },
-"proxy": { "host": "proxyhost", "port": 9018, "auth": { "username": "proxyuser", "password": "proxypass" } } } }' java -jar spring-cloud-dataflow-server-{project-version}.jar
+$ SPRING_APPLICATION_JSON='
+{
+  "maven": {
+    "local-repository": null,
+    "remote-repositories": {
+      "repo1": {
+        "url": "https://repo1",
+        "auth": {
+          "username": "repo1user",
+          "password": "repo1pass"
+        }
+      },
+      "repo2": {
+        "url": "https://repo2"
+      }
+    },
+    "proxy": {
+      "host": "proxyhost",
+      "port": 9018,
+      "auth": {
+        "username": "proxyuser",
+        "password": "proxypass"
+      }
+    }
+  }
+}
+'
 ----
 
+=== Wagon
+There is a limited support for using `Wagon` transport with maven. Currently this
+exists to support _preemptive_ authentication with `http` based repositories
+and needs to be enabled manually.
+
+Wagon based http transport is enable by setting `maven.use-wagon` property
+to `true` and then _preemptive_ authentication authentication can be enabled
+per remote repository. Configuration loosely follows similar patterns found
```

Comment:
The doc says it can be enabled `per remote repository` but we have the property at the top level as `maven.use-wagon`